src/utils/generator.js

```diff
@@ -90,13 +90,19 @@ export class Generator {
     this.realm = realm;
     this._entries = [];
     this.id = realm.nextGeneratorId++;
+    this._name = name;
   }
 
   realm: Realm;
   _entries: Array<GeneratorEntry>;
   preludeGenerator: PreludeGenerator;
   parent: void | Generator;
   id: number;
+  _name: void | string;
+
+  getName(): string {
+    return this._name || `#${this.id}`;
```

Comment:
Which generators don't have names? Just wondering if we really need the parallel id-concept that is only surfaced sometimes.

Reply:
There are many temp generators created and throw away, they do not have names yet. Also, there are sub-generators created during fork execution will not have names yet so falling back to id is safer.